mne_connectivity/spectral/time.py

```diff
@@ -639,6 +816,96 @@ def _pairwise_con(w, psd, x, y, method, kernel, foi_idx,
+        Time-frequency data.
+    x : int
+        Channel index.
+    y : int
+        Channel index.
+    method : str
+        Connectivity method.
+    kernel : array_like, shape (n_sm_fres, n_sm_times)
+        Smoothing kernel.
+    foi_idx : array_like, shape (n_foi, 2)
+        Upper and lower bound indices of frequency bands.
+    faverage : bool
+        Average over frequency bands.
+    weights : array_like, shape (n_tapers, n_freqs, n_times) | None
+        Multitaper weights.
+
+    Returns
+    -------
+    scores : list
+        List of connectivity scores between seed and target signals for each
+        connectivity method. Each element is an array with shape (n_freqs,) or
+        (n_fbands) depending on ``faverage``.
+
+    patterns : list
+        List of connectivity patterns between seed and target signals for each
+        connectivity method. Each element is an array of length 2 corresponding
+        to the seed and target patterns, respectively, each with shape
+        (n_channels, n_freqs,) or (n_channels, n_fbands) depending on
+        ``faverage``.
+    """
+    csd = []
+    for x in signals_use:
+        for y in signals_use:
+            w_x, w_y = w[x], w[y]
+            if weights is not None:
+                s_xy = np.sum(weights * w_x * np.conj(weights * w_y), axis=0)
+                s_xy = s_xy * 2 / (weights * np.conj(weights)).real.sum(axis=0)
+            else:
+                s_xy = w_x * np.conj(w_y)
+                s_xy = np.squeeze(s_xy, axis=0)
+            csd.append(_smooth_spectra(s_xy, kernel).mean(axis=-1))
+    csd = np.array(csd)
+
+    # initialise connectivity estimators and add CSD information
+    conn_class = {'mic': _MICEst, 'mim': _MIMEst, 'gc': _GCEst,
+                  'gc_tr': _GCTREst}
+    conn = []
+    for m in method:
+        # N_CONS = 1 UNTIL RAGGED ARRAYS SUPPORTED
+        call_params = {'n_signals': len(signals_use), 'n_cons': 1,
+                       'n_freqs': csd.shape[1], 'n_times': 0,
+                       'n_jobs': n_jobs}
+        if m in _gc_methods:
+            call_params['n_lags'] = gc_n_lags
+        con_est = conn_class[m](**call_params)
+        for con_i, con_csd in enumerate(csd):
+            con_est.accumulate(con_i, con_csd)
+        conn.append(con_est)
+
+    # compute connectivity
+    scores = []
+    patterns = []
+    for con_est in conn:
+        con_est.compute_con(np.array([source_idx, target_idx]), rank)
+        scores.append(con_est.con_scores[..., np.newaxis])
+        patterns.append(con_est.patterns)
+        if patterns[-1] is not None:
+            patterns[-1] = patterns[-1][..., np.newaxis]
+
+    for i, _ in enumerate(scores):
+        # mean inside frequency sliding window (if needed)
+        if isinstance(foi_idx, np.ndarray) and faverage:
+            scores[i] = _foi_average(scores[i], foi_idx)
+            if patterns[i] is not None:
+                patterns[i] = _foi_average(patterns[i], foi_idx)
```

Comment:
Can you add a unit-test for this case?

Reply:
Done! https://github.com/tsbinns/mne-connectivity/blob/894c42e59cef9f97aa07a7a4f71cef756512c50c/mne_connectivity/spectral/tests/test_spectral.py#L1050-L1088